agent/cache/cache.go

```diff
@@ -80,6 +81,8 @@ type Cache struct {
 	stopped uint32
 	// stopCh is closed when Close is called
 	stopCh chan struct{}
+	// options includes a per Cache Rate limiter specification to avoid performing too many queries
+	options *Options
```

Comment:
I do think a non-pointer `Options` would be better (similar to the constructor for the cache), but it's minor and easy enough to change later.

Reply:
Done in 48e50fe1a119b0b9480eb7c81154ff44a517d9f9